internal/server/server.go

```diff
@@ -76,11 +114,15 @@ func New(metadataStore Storer, objectStore StorerGetter, cfg *config.Config, sql
 			playground.Handler("Archivista", "/v1/query"),
 		)
 	}
-	r.PathPrefix("/swagger/").Handler(httpSwagger.WrapHandler)
-	s.router = r
 
-	return *s
+	if cfg.EnableWitnessDistro {
+		r.HandleFunc("/v1/witness/", s.WitnessAllVersionsHandler)
+		r.HandleFunc("/v1/witness/{version}", s.WitnessVersionHandler)
+		r.HandleFunc("/v1/witness/{version}/{distribution}", s.DownloadWitnessHandler)
+	}
```

Comment:
Probably not needed for now but, we should consider how to upload/update for new versions of Witness. Depending on where Archivista is running, it could be difficult to access it's storage to update config files or add witness binaries. Also, updating config file will require a file system watcher or a restart of archivista.

Reply:
Yeah, this is very true on all accounts